token_import.py

```diff
@@ -46,4 +24,27 @@ def read_used_list():
         print(f"{USED_FILE} not exist, create one!")
         f = open(USED_FILE, 'w+')
         f.close()
-        return list()
\ No newline at end of file
+        return list()
+
+def _import_qrcode_from_bigquery(token_dict: dict, table_name: str) -> dict:
+    current_year = datetime.now().year
+    query = f"""
+        SELECT
+          ticket_type,
+          qrcode
+        FROM
+          `pycontw-225217.{table_name}`
+        WHERE
+          qrcode IS NOT NULL
+          AND payment_status = 'paid'
+          AND EXTRACT(YEAR
+          FROM
+            CAST(paid_date AS DATE)) = {current_year};
+    """
+    query_job = BIQUERY_CLIENT.query(query)
+    for row in query_job:
+        ticket_name = row["ticket_type"].lower().replace("\"", "")
+        qrcode = row["qrcode"].replace("\"", "")
+        assert len(qrcode) == 32, f"qrcode length should be 32, but got {len(qrcode)}"
```

Comment:
Want to confirm if this line is your intention, cuz I think this will throw AssertionError even if there is only one `qrcode` has length other than 32?

Reply:
yea good point, let me change it into sending alerts to discord instead